src/App.ts

```diff
@@ -1070,6 +1070,11 @@ export default class App {
     appToken?: string,
     logger?: Logger,
   ): Receiver {
+    if (port !== undefined && receiver !== undefined) {
+      // Check https://github.com/slackapi/bolt-js/issues/1357 for the context
+      this.logger.info("You've passed both port and receiver. Some receivers may not respect the port passed in App constructor. If you encounter the unexpected behavior, check the appropriate way to pass port to the receiver.");
```

Comment:
```suggestion
      this.logger.info("You provided both port and receiver arguments. Some receivers may not respect the port passed to the App constructor. If you encounter any unexpected behavior, check the appropriate way to pass port to the receiver.");
```

Reply:
So the above is just some grammar and wording tweaking, however, can we do anything more to help people potentially hitting this issue? In particular, my thinking on what I would feel/do if I encountered this error:

- "Some receivers" may not respect the port passed. Can we be more specific? Which receivers do not respect it? Can we inspect the receiver class instance, for example, and provide more specific information?
- For the final sentence, "check the appropriate way to pass the port." From where? Should we link to a document?
- Should this be a `info` or a `warn` message?

I know this might seem very specific but I think I would be annoyed with the level of ambiguity communicated in this message if I were to receive it as a user 😞

Reply:
@filmaj 
>"Some receivers" may not respect the port passed. Can we be more specific? Which receivers do not respect it? Can we inspect the receiver class instance, for example, and provide more specific information?

Does "Your receiver (or this receiver) may not respect the port passed to the App constructor." sound clearer to you? I don't see a big difference here, though.

>For the final sentence, "check the appropriate way to pass the port." From where? Should we link to a document?

This depends on whether the receiver is from us or a custom one. If we should be more specific here, you can say "check the receiver code to learn how it treats the port argument passed to the App constructor."

Currently, even the built-in ones have this issue. Instead of having this logging, we may should resolve the unexpected behavior. What do you think?

>Should this be a info or a warn message?

I don't think that this should be warn level because the receiver an app uses may not have the issue.

Reply:
The only thing I think we could improve is clarifying "some receivers." It looks like the AWSReceiver and the SocketModeReceiver _without_ OAuth or custom routes do not use the port parameter for anything. So perhaps we can be specific by mentioning those two situations in the log?

Reply:
@filmaj Thanks for your reply!

>Currently, even the built-in ones have this issue. Instead of having this logging, we may should resolve the unexpected behavior. What do you think?

Now I'm leaning towards fixing the issue instead of trying to have an accurate logging for the scenarios you mentioned. 

I will close this PR that just adds logging now. We may work on the improvement for the issue #1357 sometime in the future when we have the bandwidth for it.